### PR TITLE
RATIS-840. memory leak of log appender

### PR DESCRIPTION
**Pull request by runzhiwang**
refer it [RATIS-840](https://issues.apache.org/jira/browse/RATIS-840) for details

### CODE REVIEW

**Comment by runzhiwang** [comment]
@szetszwo @lokeshj1703 @bshashikant 
I think this patch does not generate new failed UT.
I trigger CI several times, all the [failed UT](https://github.com/runzhiwang/incubator-ratis/runs/628048866) are in the list of [RATIS-863](https://issues.apache.org/jira/browse/RATIS-863).  I together with @bshashikant summary all the failed UT generated by ratis/master in RATIS-863.

Besides, the failed UT in this PR has nothing to do with this patch. You can find the reason in [RATIS-912](https://issues.apache.org/jira/browse/RATIS-912)

File: ratis-test/src/test/java/org/apache/ratis/grpc/TestLogAppenderWithGrpc.java [hunk 1]
```diff
@@ -124,8 +124,5 @@ private void runTestRestartLogAppender(MiniRaftClusterWithGrpc cluster) throws E
         Assert.assertTrue(reply.isSuccess());
       }
     }
-
-    // assert INCONSISTENCY counter should become 1
-    Assert.assertEquals(1L, leaderMetrics.getRegistry().counter(counter).getCount());
```
Comment (lokeshj1703) [inline ratis-test/src/test/java/org/apache/ratis/grpc/TestLogAppenderWithGrpc.java:129]: I think we can assert that leaderMetrics.getRegistry().counter(counter).getCount() >= 1?

Comment (runzhiwang) [reply ratis-test/src/test/java/org/apache/ratis/grpc/TestLogAppenderWithGrpc.java]: @lokeshj1703  Thanks for review. I have updated.

**Comment by lokeshj1703** [comment]
Let's wait for @szetszwo to take a look before committing.

**Comment by szetszwo** [comment]
+1 the change looks good.

**Pull request closed without merging**